File: src/atlas.ui/src/app/tones.ts
import type {
  GrowthGoal,
  GrowthGoalActionState,
  GrowthGoalCheckInSignal,
  HealthSignal,
  Priority,
  Risk,
  RiskStatus,
  TaskStatus,
} from './types'

export function taskStatusTone(s?: TaskStatus) {
  if (s === 'Done') return 'toneGood'
  if (s === 'Blocked') return 'toneBad'
  if (s === 'In Progress') return 'toneInfo'
  return 'toneNeutral'
}

export function priorityTone(p?: Priority) {
  if (!p || p === 'Low') return 'toneNeutral'
  if (p === 'Medium') return 'toneWarn'
  return 'toneBad'
}

export function riskStatusTone(status: RiskStatus) {
  if (status === 'Open') return 'toneBad'
  if (status === 'Watching') return 'toneWarn'
  if (status === 'Resolved') return 'toneGood'
  return 'toneNeutral'
}

export function severityTone(sev: Risk['severity']) {
  if (sev === 'High') return 'toneBad'
  if (sev === 'Medium') return 'toneWarn'
  return 'toneNeutral'
}

export function healthTone(health?: HealthSignal) {
  if (health === 'Green') return 'toneGood'
  if (health === 'Yellow') return 'toneWarn'
  if (health === 'Red') return 'toneBad'
  return 'toneNeutral'
}

export function projectStatusTone(status?: string) {
  if (status === 'Active') return 'toneGood'
  if (status === 'Paused') return 'toneWarn'
  return 'toneNeutral'
}

export function goalStatusTone(status: GrowthGoal['status']) {
  if (status === 'Completed' || status === 'OnTrack') return 'toneGood'
  if (status === 'NeedsAttention') return 'toneWarn'
  return 'toneNeutral'
}

export function actionStateTone(state: GrowthGoalActionState) {
  if (state === 'Complete') return 'toneGood'
  if (state === 'InProgress') return 'toneWarn'
  return 'toneNeutral'
}

export function checkInSignalTone(signal: GrowthGoalCheckInSignal) {
  if (signal === 'Positive') return 'toneGood'
  if (signal === 'Mixed') return 'toneWarn'
  if (signal === 'Concern') return 'toneBad'
  return 'toneNeutral'
}

export function signalTone(value: string) {
  const v = value.toLowerCase()
  if (v.includes('blocked')) return 'toneBad'
  if (v.includes('atrisk') || v.includes('heavy') || v === 'high' || v.includes('medium')) return 'toneWarn'
  if (v.includes('ontrack') || v.includes('light') || v === 'low') return 'toneGood'
  return 'toneNeutral'
}

export function ticketAttentionTone(status: string) {
  const s = status.toLowerCase()
  if (s.includes('blocked')) return 'toneBad'
  if (s.includes('code review') || s.includes('in review') || s.includes('review')) return 'toneWarn'
  return 'toneNeutral'
}

File: src/frontend/Atlas.Ui/Mapping/Tones.cs
using Atlas.Ui.Models;

namespace Atlas.Ui.Mapping;

/// <summary>CSS tone class helpers ported from React <c>app/tones.ts</c>.</summary>
public static class Tones
{
    public static string TaskStatusTone(Models.TaskStatus? status) => status switch
    {
        Models.TaskStatus.Done => "toneGood",
        Models.TaskStatus.Blocked => "toneBad",
        Models.TaskStatus.InProgress => "toneInfo",
        _ => "toneNeutral"
    };

    public static string PriorityTone(Priority? priority)
    {
        if (priority is null or Priority.Low)
        {
            return "toneNeutral";
        }

        if (priority is Priority.Medium)
        {
            return "toneWarn";
        }

        return "toneBad";
    }

    public static string RiskStatusTone(RiskStatus status) => status switch
    {
        RiskStatus.Open => "toneBad",
        RiskStatus.Watching => "toneWarn",
        RiskStatus.Resolved => "toneGood",
        _ => "toneNeutral"
    };

    public static string SeverityTone(string? severity) => severity switch
    {
        "High" => "toneBad",
        "Medium" => "toneWarn",
        _ => "toneNeutral"
    };

    public static string HealthTone(HealthSignal? health) => health switch
    {
        HealthSignal.Green => "toneGood",
        HealthSignal.Yellow => "toneWarn",
        HealthSignal.Red => "toneBad",
        _ => "toneNeutral"
    };

    public static string ProjectStatusTone(string? status) => status switch
    {
        "Active" => "toneGood",
        "Paused" => "toneWarn",
        _ => "toneNeutral"
    };

    public static string GoalStatusTone(GrowthGoalStatus status) => status switch
    {
        GrowthGoalStatus.Completed or GrowthGoalStatus.OnTrack => "toneGood",
        GrowthGoalStatus.NeedsAttention => "toneWarn",
        _ => "toneNeutral"
    };

    public static string ActionStateTone(GrowthGoalActionState state) => state switch
    {
        GrowthGoalActionState.Complete => "toneGood",
        GrowthGoalActionState.InProgress => "toneWarn",
        _ => "toneNeutral"
    };

    public static string CheckInSignalTone(GrowthGoalCheckInSignal signal) => signal switch
    {
        GrowthGoalCheckInSignal.Positive => "toneGood",
        GrowthGoalCheckInSignal.Mixed => "toneWarn",
        GrowthGoalCheckInSignal.Concern => "toneBad",
        _ => "toneNeutral"
    };

    public static string SignalTone(string value)
    {
        var v = value.ToLowerInvariant();
        if (v.Contains("blocked"))
        {
            return "toneBad";
        }

        if (v.Contains("atrisk") || v.Contains("heavy") || v == "high" || v.Contains("medium"))
        {
            return "toneWarn";
        }

        if (v.Contains("ontrack") || v.Contains("light") || v == "low")
        {
            return "toneGood";
        }

        return "toneNeutral";
    }

    public static string TicketAttentionTone(string status)
    {
        var s = status.ToLowerInvariant();
        if (s.Contains("blocked"))
        {
            return "toneBad";
        }

        if (s.Contains("code review") || s.Contains("in review") || s.Contains("review"))
        {
            return "toneWarn";
        }

        return "toneNeutral";
    }
}
